backend/dataall/db/api/dataset.py

```diff
@@ -491,12 +500,12 @@ def list_dataset_shares(session, dataset_uri) -> [models.ShareObject]:
         return query.all()
 
     @staticmethod
-    def list_dataset_approved_shares(session, dataset_uri) -> [models.ShareObject]:
+    def list_dataset_shares_with_existing_shared_items(session, dataset_uri) -> [models.ShareObject]:
         query = session.query(models.ShareObject).filter(
             and_(
                 models.ShareObject.datasetUri == dataset_uri,
                 models.ShareObject.deleted.is_(None),
-                models.ShareObject.status == 'Approved',
+                models.ShareObject.existingSharedItems.is_(True),
```

Comment:
This call does not work in my opinion as existingSharedItems is a query expression which need to be defined, see https://docs.sqlalchemy.org/en/14/orm/mapped_sql_expr.html#query-time-sql-expressions-as-mapped-attributes

Instead we could join here with the ShareObjectItem table to verify if shared items exist.